dt_model/symbols/context_variable.py

```diff
@@ -1,23 +1,34 @@
+"""
+This module defines context variables. A context variable models a model variable
+that is not directly controlled by the model, but that can influence the model
+behavior. In general, context variables are sampled from a distribution, either
+categorical or continuous.
+"""
+
 from __future__ import annotations
 
 import random
+from abc import ABC, abstractmethod
 
 from scipy.stats import rv_continuous
 
-from dt_model.symbols._base import SymbolExtender
+from ..engine.frontend import graph
+from ..internal.sympyke.symbol import SymbolValue
 
 
-class ContextVariable(SymbolExtender):
+class ContextVariable(ABC):
```

Comment:
Note to self: here we could easily use a `Protocol` (an empty `ABC` is basically a `Protocol`).